File: nijesmik/week10/호텔_방_배정.js
function solution(k, room_number) {
  const taken = new Map();
  return room_number.map((num) => get(taken, num));
}

const get = (taken, num) => {
  if (!taken.has(num)) {
    taken.set(num, num + 1);
    return num;
  }
  const val = get(taken, taken.get(num));
  taken.set(num, val);
  return val;
};

File: Minah7/week10/호텔 방 배정.java
// 정확성 성공, 효율성 실패.

import java.util.*;

class Solution {
    public long[] solution(long k, long[] room_number) {
        long[] answer = new long[room_number.length];
        Set<Long> set = new HashSet<>();
        for(int i = 0; i < room_number.length; i++) {
            long room = room_number[i];
            if(!set.contains(room)) {
                set.add(room);
                answer[i] = room;
            } else {
                for(long start = room + 1; start <= k; start++) {
                    if(!set.contains(start)) {
                        set.add(start);
                        answer[i] = start;
                        break;
                    }
                }
            }
        }
        return answer;
    }
}